jaxrs/src/main/java/se/fortnox/reactivewizard/jaxrs/params/deserializing/CustomClassDeserializerFactory.java

```diff
@@ -0,0 +1,92 @@
+package se.fortnox.reactivewizard.jaxrs.params.deserializing;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+
+class CustomClassDeserializerFactory {
+
+    private CustomClassDeserializerFactory() {
+    }
+
+    @Nullable
+    public static <T> Deserializer<T> createOrNull(Class<T> cls) {
+        var methods = cls.getMethods();
+
+        var valueOfDeserializer = findStaticMethodDeserializer(methods, "valueOf", cls);
```

Comment:
Optional can be used to skip the nullchecks here if you wish, but it's not necessary.

```
return ofNullable(findStaticMethodDeserializer(methods, "valueOf", cls))
			.or(() -> ofNullable(findStaticMethodDeserializer(methods, "fromString", cls)))
			.or(() -> ofNullable(findConstructorDeserializer(cls)))
			.orElse(null);
```

Reply:
I'll let you decide. I personally slightly prefer the null-checks in this scenario.